htlcswitch/link.go

```diff
@@ -1356,6 +1369,23 @@ func (l *channelLink) handleDownStreamPkt(pkt *htlcPacket, isReProcess bool) {
 		pkt.outgoingHTLCID = index
 		htlc.ID = index
 
+		l.cfg.HTLCNotifier.NotifyForwardingEvent(htlcnotifier.ForwardingEvent{
```

Comment:
Think it will be ok to export `htlcPacket`? Otherwise `htlcnotifier` needs to be moved into `htlcSwitch`.  Nice idea though, these do get a bit lenghty!